### ISSUE TITLE
Tracking issue for highlighted diffs

### ISSUE THREAD

**Issue by dtolnay** [dtolnay/trybuild#41]
Requires opt-in for now (`cargo test --features trybuild/diff`).

Currently unix-only. We will need to look into how to get highlighted background color reliably on Windows terminals, and ideally implement support for it in https://github.com/BurntSushi/termcolor.

Our diff algorithm is backed by https://github.com/dtolnay/dissimilar which is intended to produce human readable diffs with less noise than raw Myers' algorithm.

The intended output looks like:

<p align="center"><img src="https://user-images.githubusercontent.com/1940490/71450848-e365b700-2737-11ea-9656-116ef72c01b9.png" width="80%"></p>

**Comment by CreepySkeleton** [comment]
@dtolnay Would it be possible to introduce `TRYBUILD_DIFF=auto/always/never` environment variable? Sometimes (quite often) the diff is not shown but I need it.

**Comment by dtolnay** [comment]
Can you give an example output when you need a diff but it is not shown? There are heuristics that determine when a diff is worth showing and we can adjust those.

**Comment by CreepySkeleton** [comment]
Expected
```text
error: parse must have exactly one argument
  --> $DIR/parse_not_zero_args.rs:14:17
   |
14 |     #[structopt(parse(from_str, from_str))]
```
Actual
```text
error: `parse` must have exactly one argument
  --> $DIR/parse_not_zero_args.rs:14:17
   |
14 |     #[structopt(parse(from_str, from_str))]

**Comment by dtolnay** [comment]
Looks fine to me.

<p align="center"><img src="https://user-images.githubusercontent.com/1940490/73602244-7e1a9800-4525-11ea-8844-7a8a2006d758.png" width="70%"></p>

**Comment by CreepySkeleton** [comment]
Doesn't work in Git Bash for windows 
![image](https://user-images.githubusercontent.com/50968528/73602456-30ce1300-4585-11ea-9d74-03584a028b94.png)

`trybuild 1.0.21`

**Comment by dtolnay** [comment]
Yep, it says at the top of the issue this is currently unix-only. Termcolor would need a highlighting implementation for windows before I would try to make it work there.

**Comment by pksunkara** [comment]
IIRC termcolor has support for windows and we used that for clap. Not sure what the issue here is.
